lib/SDIR/Translate/translateToSDFG.cpp

```diff
@@ -0,0 +1,1597 @@
+
+  /*if (GetAccessOp aNode = dyn_cast<GetAccessOp>(load.arr().getDefiningOp())) {
+    jemit.printKVPair("data", aNode.getName());
+  } else {
+    return failure();
+  }*/
+
+  /*jemit.startNamedObject("subset");
+  jemit.printKVPair("type", "Range");
+  jemit.startNamedList("ranges");
+
+  if(ArrayAttr syms = load->getAttr("indices").cast<ArrayAttr>()){
+    for(Attribute sym : syms.getValue()){
+      if (StringAttr sym_str = sym.cast<StringAttr>()) {
+        jemit.startObject();
+        jemit.printKVPair("start", sym_str.getValue());
+        jemit.printKVPair("end", sym_str.getValue());
+        jemit.printKVPair("step", 1);
+        jemit.printKVPair("tile", 1);
+        jemit.endObject();
+      } else {
+        return failure();
+      }
+    }
+  } else {
+    return failure();
+  }
+
+  jemit.endList(); // ranges
+  jemit.endObject(); // subset
+  */
+
+  /*jemit.startNamedObject("src_subset");
+  jemit.printKVPair("type", "Range");
+  jemit.startNamedList("ranges");
+
+  if(ArrayAttr syms = load->getAttr("indices").cast<ArrayAttr>()){
+    for(Attribute sym : syms.getValue()){
+      if (StringAttr sym_str = sym.cast<StringAttr>()) {
+        jemit.startObject();
+        jemit.printKVPair("start", sym_str.getValue());
+        jemit.printKVPair("end", sym_str.getValue());
+        jemit.printKVPair("step", 1);
+        jemit.printKVPair("tile", 1);
+        jemit.endObject();
+      } else {
+        return failure();
+      }
+    }
+  } else {
+    return failure();
+  }
+
+  jemit.endList(); // ranges
+  jemit.endObject(); // src_subset
+  */
+
+  jemit.endObject(); // attributes
+  jemit.endObject(); // data
+  jemit.endObject(); // attributes
+
+  jemit.printKVPair("src", taskSrc.ID());
+  jemit.printKVPair("src_connector", "__out");
+
+  jemit.printKVPair("dst", taskDest.ID());
+
+  std::string argname;
+  AsmState state(taskDest);
+  BlockArgument bArg = taskDest.getArgument(argIdx);
+  llvm::raw_string_ostream argnameStream(argname);
+  bArg.printAsOperand(argnameStream, state);
+
+  argname.erase(0, 1); // remove %-sign
+  jemit.printKVPair("dst_connector", argname);
+
+  jemit.endObject();
+  return success();
+}
+
+LogicalResult translateCallToSDFG(sdir::CallOp &op, JsonEmitter &jemit) {
+  if (op.callsTasklet()) {
+    TaskletNode task = op.getTasklet();
+    for (unsigned i = 0; i < op.getNumOperands(); ++i) {
+      Value val = op.getOperand(i);
+      if (LoadOp load = dyn_cast<LoadOp>(val.getDefiningOp())) {
+        if (printLoadTaskletEdge(load, task, i, jemit).failed())
+          return failure();
+      } else if (sdir::CallOp call =
+                     dyn_cast<sdir::CallOp>(val.getDefiningOp())) {
+        TaskletNode taskSrc = call.getTasklet();
+        if (printTaskletTaskletEdge(taskSrc, task, i, jemit).failed())
+          return failure();
+      } else if (arith::ConstantOp con =
+                     dyn_cast<arith::ConstantOp>(val.getDefiningOp())) {
+        // TODO: Add constants
```

Comment:
necessary?

Reply:
Yes if we want to support snippets like this:

```
%1 = arith.constant 1 : i32
%c = sdir.call @add(%1, %1) : (i32, i32) -> i32
```

Reply:
Yes if we want to support snippets like this:

```
%1 = arith.constant 1 : i32
%c = sdir.call @add(%1, %1) : (i32, i32) -> i32
```

Reply:
let's discuss it in the "store" discussion.

Reply:
@Berke-Ates If you resolve this with the elegant solution from the `store` discussion, it will resolve this one too.